Source/DafnyLanguageServer/Workspace/DocumentDatabase.cs

```diff
@@ -99,65 +109,107 @@ ISymbolTableRelocator symbolTableRelocator
       }
       databaseEntry.CancelPendingUpdates();
       var cancellationSource = new CancellationTokenSource();
+      var previousDocumentTask = FirstSuccessfulTask(databaseEntry.VerifiedDocument, databaseEntry.ResolvedDocument);
+      var resolvedDocumentTask = ApplyChangesAsync(previousDocumentTask, documentChange, cancellationSource.Token);
+      var verifiedDocument = VerifyAsync(resolvedDocumentTask, VerifyOnChange, cancellationSource.Token);
       var updatedEntry = new DocumentEntry(
         documentChange.TextDocument.Version,
-        Task.Run(async () => await ApplyChangesAsync(await databaseEntry.Document, documentChange, cancellationSource.Token)),
+        resolvedDocumentTask,
+        verifiedDocument,
         cancellationSource
       );
       documents[documentUri] = updatedEntry;
-      return await updatedEntry.Document;
+      return resolvedDocumentTask.ToObservable().Concat(verifiedDocument.ToObservable());
+    }
+
+
+#pragma warning disable VSTHRD200
+    private static Task<T> FirstSuccessfulTask<T>(params Task<T>[] tasks) {
```

Comment:
Please rename this method `FirstSuccessfulAsync<T>` to get rid of the warning removal directives.